handlers/upload.go

```diff
@@ -86,6 +87,15 @@ func (r UploadVODRequest) getTargetURL() (*url.URL, error) {
 	return nil, fmt.Errorf("no output_location with transcoded_segments")
 }
 
+func (r UploadVODRequest) isAutoMP4() bool {
+	for _, o := range r.OutputLocations {
+		if o.Outputs.AutoMP4 {
+			return true
```

Comment:
WDYT of returning the location of the `URL` of the MP4 location instead and saving it on the args object? Otherwise we're completely ignoring it here and assuming it has to be the same location as the HLS playlist etc.

If it requires too much changes in the core logic (like splitting the logic to copy the files etc which might be cumbersome), another idea would be returning this "is auto mp4" boolean from the `getTargetURL` function instead. So we explicitly only support the same output location for both locations, and just return whether the "target url" location has `auto_mp4` turned on.

Reply:
yeah i was considering doing the latter tbh, since we do currently only support a single output location. i'll go with that